woocommerce-services.php

```diff
@@ -530,13 +530,29 @@ public function init_core_wizard_shipping_config() {
 
 		public function init_core_wizard_payments_config() {
 			$stripe_settings = get_option( 'woocommerce_stripe_settings', false );
-			$stripe_enabled  = is_array( $stripe_settings )
+
+			$user_elected_to_create_stripe_account  = is_array( $stripe_settings )
 				&& ( isset( $stripe_settings['create_account'] ) && 'yes' === $stripe_settings['create_account'] )
 				&& ( isset( $stripe_settings['enabled'] ) && 'yes' === $stripe_settings['enabled'] );
 
-			if ( $stripe_enabled && is_plugin_active( 'woocommerce-gateway-stripe/woocommerce-gateway-stripe.php' ) ) {
+			// In certain scenarios, the user can enter an email address but not connect Jetpack in the wizard,
+			// but instead add the Stripe keys manually and connect Jetpack after. If the existing keys are detected,
+			// forget the wizard settings and never retry
+			$stripe_already_connected = is_array( $stripe_settings )
+				&& (
+					! empty( $stripe_settings['test_publishable_key'] )
+					|| ! empty( $stripe_settings['test_secret_key'] )
+					|| ! empty( $stripe_settings['publishable_key'] )
+					|| ! empty( $stripe_settings['secret_key'] )
+				);
+			if ( $user_elected_to_create_stripe_account && $stripe_already_connected ) {
+				unset( $stripe_settings['email'] );
 				unset( $stripe_settings['create_account'] );
 				update_option( 'woocommerce_stripe_settings', $stripe_settings );
+				return;
+			}
+
+			if ( $user_elected_to_create_stripe_account && is_plugin_active( 'woocommerce-gateway-stripe/woocommerce-gateway-stripe.php' ) ) {
 				$this->tracks->record_user_event( 'core_wizard_stripe_setup' );
```

Comment:
The `core_wizard_stripe_setup` is getting tracked before we check for an error in the response below. Should we track it after we know the request was successful? If we decide to track the failure above, we could track the failure here as well.